frontend/src/layout/header/UserBox.tsx

```diff
@@ -107,149 +171,189 @@ const WithFloatingMenu: React.FC<{ children: ReactElement }> = ({ children }) =>
                 {user.canUpload && <MenuItem
                     icon={<FiUpload />}
                     linkTo={"/~upload"}
                     onClick={() => close()}
                 >{t("upload.title")}</MenuItem>}
-                <MenuItem
-                    icon={<FiFilm />}
-                    linkTo="/~manage"
-                    onClick={() => close()}
-                >{t("user.manage-content")}</MenuItem>
             </>}
 
-            <MenuItem icon={<HiOutlineTranslate />} onClick={() => setState("language")}>
-                {t("language")}
-            </MenuItem>
-
             {/* Logout button if the user is logged in */}
             {isRealUser(user) && <Logout />}
         </>,
         language: () => <>
-            <MenuItem icon={<FiChevronLeft />} onClick={() => setState("main")} borderBottom>
-                {t("back")}
-            </MenuItem>
-            <LanguageMenu />
+            <ReturnButton onClick={() => close()}>{t("Choose language")}</ReturnButton>
+            <LanguageMenu close={() => close()} />
         </>,
     });
 
-    return (
-        <Floating padding={0}>
+    return <Floating borderWidth={0} padding={0}>
+        <div
+            onClick={e => {
+                if (e.target === e.currentTarget) {
+                    close();
+                }
+            }}
+            onBlur={handleBlur}
+            css={{
+                position: "relative",
+                // Grey out background on mobile devices.
+                [`@media (max-width: ${BREAKPOINT_MEDIUM}px)`]: {
+                    position: "fixed",
+                    top: "var(--header-height)",
+                    bottom: 0,
+                    left: 0,
+                    right: 0,
+                    zIndex: 1001,
+                    backgroundColor: "#000000a0",
+                },
+            }}>
             <ul css={{
-                listStyle: "none",
+                borderRadius: 8,
+                zIndex: 1000,
+                right: 0,
                 margin: 0,
-                paddingLeft: 0,
                 minWidth: 200,
+                paddingLeft: 0,
+                overflow: "hidden",
+                listStyle: "none",
+                [`@media (max-width: ${BREAKPOINT_MEDIUM}px)`]: {
+                    backgroundColor: "white",
+                    marginTop: 0,
+                    position: "fixed",
+                    left: 0,
+                    top: 0,
+                    boxShadow: "none",
+                    borderRadius: "0 0 8px 8px",
+                },
             }}>{items}</ul>
-        </Floating>
-    );
+        </div>
+    </Floating>;
+};
+
+
+const LanguageSettings: React.FC = () => {
+    const { t } = useTranslation();
+
+    return <WithFloatingMenu type="language">
+        <ActionIcon title={t("language")} css={{
+            [`@media not all and (max-width: ${BREAKPOINT_SMALL}px)`]: {
+                button: { padding: 7 },
+            },
```

Comment:
Is that intended? Seems fine to me without?

Reply:
It is intended, as the icon uses a fixed size of 24px. The other icons have a size of 28px on screen sizes with a width  above 450, and 24px below that. That means that on larger screens, in order for all buttons to have the same size (mainly for hover/focus purposes), the language icon needs some more padding (2px in each direction, to be exact).